### DISCUSSION TITLE
Force no punctuation

### DISCUSSION THREAD

**Discussion by razem-io** [openai/whisper#589] [Q&A] [answered]
Hi, is it possible to force whisper to not use punctuation at all? I would like to do this manually in post-processing. I have the following problem. When I say:

"Hello comma nice to meet you exclamation mark"

Whisper most of the time does this then:

"Hello comma, nice to meet you exclamation mark."

In post, I remove all the punctuation (",", ".") that whisper did automatically and replace the manual punctuation. It looks like this then:

"Hello, nice to meet you!"

However, sometimes whisper replaces "comma" with "," which sucks for my manual punctuation because now I can not decide if it was set automatically or not.

**Comment by jongwook** [comment, marked as answer]
Common punctuation symbols like commas, periods, exclamation points, etc. are not included in the default list of symbols that are suppressed during decoding:

https://github.com/openai/whisper/blob/eff383b27b783e280c089475852ba83f20f64998/whisper/tokenizer.py#L248

You could try adding them back in this list to obtain output without any of those punctuation symbols.

**Reply by razem-io** [reply]
Thank you very much. I will try that!

**Reply by alexanderbluhm** [reply]
@razem-io Did you have any luck with this?

**Reply by razem-io** [reply]
I haven't tried it yet. I resulted to delete all symbols and use special keywords for all punctuation. This works extremely well but requires some training. After whisper I also run LanguageTool.org (open source version) to only fix punctuation. In combination this is the perfect solution for me.

**Comment by ryanheise** [comment]
You can use `--suppress_tokens` to specify all the punctuation symbols to suppress. You need to specify the numeric token "ids", not the text. Using the tokenizer, I found that "!" is 0, "," is 11, "." is 13 and "?" is 30. So if those are the only punctuation symbols you care about, you could run:

```
whisper --suppress_tokens=0,11,13,30 yourfile.mp3
```

**Reply by razem-io** [reply]
Oh wow, thank you for this piece of information. Was to stupid to understand that. I literally put in the symbols instead of the numbers.

**Reply by Rumeysakeskin** [reply]
> You can use `--suppress_tokens` to specify all the punctuation symbols to suppress. You need to specify the numeric token "ids", not the text. Using the tokenizer, I found that "!" is 0, "," is 11, "." is 13 and "?" is 30. So if those are the only punctuation symbols you care about, you could run:
> 
> ```
> whisper --suppress_tokens=0,11,13,30 yourfile.mp3
> ```

Hello @ryanheise ,
How can I reach all the symbols?
Thanks in advance.

**Reply by ryanheise** [reply]
Using python code:

```
from whisper.tokenizer import get_tokenizer

is_multilingual = True
language = "en"

tokenizer = get_tokenizer(is_multilingual, language=language, task="transcribe")
print(tokenizer.encode("?"))
print(tokenizer.encode("!"))
print(tokenizer.encode("."))
```

That prints the token ID for the "?", "!" and "." symbols. You can change the parameters depending on the language and whether or not you're using a multilingual model.

(If you're not a Python coder, just run python and type this in)

**Comment by 1001111github** [comment]
Thank you all, preparing Whisper output for non-AI computer input. This page was invaluable.
